python/dgl/contrib/sampling/sampler.py

```diff
@@ -530,10 +530,10 @@ class EdgeSampler(object):
     edge only if the triple (source node, destination node and relation)
     matches one of the edges in the graph.
 
-    For uniform sampling, the sampler generates only num_of_edges/batch_size 
-    samples. 
-
-    For uniform sampling, the sampler generates samples infinitly.
+    This sampler samples edges without replacement by default, which means it 
+    returns a fixed number of batches (i.e., num_edges/batch_size). However 
+    it accepts a replacement argument, which can be set to True to let the 
+    sampler generates any number of batches.
```

Comment:
it seems with or without replacement, the sampler always generate `num_edges/batch_size` batches.

Reply:
if replacement == False, it will generate num_edges/batch_sizes.
if replacement == True, It will generate samples infinitely.